sambacc/commands/config.py

```diff
@@ -141,14 +149,81 @@ def _call_if_leader(
     return _call_if_leader
 
 
+def _readpid(path: pathlib.Path) -> int:
+    try:
+        return int(path.read_text())
+    except Exception as err:
+        _logger.warning("Unable to read pid file %r: %s", path, err)
+        return -1
+
+
+def _signal_pids_dir(
+    pids_dir: str,
+    current: config.InstanceConfig,
+    previous: typing.Optional[config.InstanceConfig],
+) -> UpdateResult:
+    changed = current != previous
+    if not changed:
+        return current, changed
+
+    # something changed, send signal to pids found in pid files in dir
+    try:
+        _pids = [
+            _readpid(p)
+            for p in pathlib.Path(pids_dir).iterdir()
+            if p.name.endswith(".pid")
+        ]
+    except FileNotFoundError:
+        _logger.warning("pids dir not found: %r; skipping pids", pids_dir)
+        return current, changed
+
+    pids = [pid for pid in _pids if pid > 0]  # filter out errors
```

Comment:
Maybe change to `pid > 1` (pid number 1 is init in Linux)

Reply:
In containers pid one could be a sambacc component or smbd. It probably won't be in ceph (IIRC it should be a podman init component) but I would rather not exclude a valid but special pid because I do want the feature reusable on (theoretical) non-ceph deployments.